src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/CreateTableIntegrationTest.java

```diff
@@ -685,27 +685,58 @@ private static Stream<Arguments> allTableData() {
+                                                          {
+                                                            "name": "invalidVectorType",
+                                                            "definition": {
+                                                              "columns": {
+                                                                "id": "text",
+                                                                "age": "int",
+                                                                "name": "text",
+                                                                "vector_type": {
+                                                                  "type": "vector",
+                                                                  "dimension": "aaa"
+                                                                }
+                                                              },
+                                                              "primaryKey": "id"
+                                                            }
+                                                          }
+                                                                              """,
                   "invalidVectorType not primitive type provided",
                   true,
                   invalidVectorType.code,
                   invalidVectorType.body)));
-
+      // vector type with vectorize
+      testCases.add(
+          Arguments.of(
+              new CreateTableTestData(
+                  """
+                                    {
+                                       "name": "vectorizeConfigTest",
+                                       "definition": {
+                                           "columns": {
+                                               "id": {
+                                                   "type": "text"
+                                               },
+                                               "age": {
+                                                   "type": "int"
+                                               },
+                                               "content": {
+                                                 "type": "vector",
+                                                 "dimension": 1024,
+                                                 "service": {
+                                                   "provider": "nvidia",
+                                                   "modelName": "NV-Embed-QA"
+                                                 }
+                                               }
+                                           },
+                                           "primaryKey": "id"
+                                       }
+                                    }
+                                    """,
+                  "primaryKeyAsStringTable",
+                  false,
+                  null,
+                  null)));
```

Comment:
Pretty extensive tests -- great. But would be good to also cover invalid "service" if not (I may have missed it)? Just because of the way conversion is done (as per my earlier comment).

Reply:
Wanted to do another PR converting the the errors in validator to use Error V2 framework. fwiw will add test case using Error V1 and change them when Error V2 is implemented.